File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/user/UserRepository.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.user;

import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, String> {

    @EntityGraph(attributePaths = { "usernameUserRoles.role" })
    User findByUsernameIgnoreCase(String username);

    User findFirstByOwnerId(Integer id);
}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/user/UserRepository.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.user

import org.springframework.data.jpa.repository.JpaRepository


interface UserRepository : JpaRepository<User, String> {

    fun findByUsernameIgnoreCase(username: String): User?

    fun findFirstByOwnerId(id: Int): User?

    fun existsByUsernameIgnoreCase(username: String?): Boolean

}
